mitosheet/src/components/taskpanes/Graph/GraphSidebar.tsx

```diff
@@ -137,37 +166,90 @@ const GraphSidebar = (props: {
     */
     const [graphUpdatedNumber, setGraphUpdatedNumber] = useState(0)
 
-    const dataSourceSheetIndex = graphParams.graphCreation.sheet_index
-    const graphOutput = props.graphDataJSON[dataSourceSheetIndex]?.graphOutput
-    const [_copyGraphCode, graphCodeCopied] = useCopyToClipboard(graphOutput?.graphGeneratedCode);
-    const [stepID, setStepID] = useState<string|undefined>(undefined);
-
-    const [loading, setLoading] = useState<boolean>(false)
+    // Log when the graph has been opened
+    useEffect(() => {
+        void props.mitoAPI.log('opened_graph');
+    }, []);
 
     // Save the last step index, so that we can check if an undo occured
     const prevLastStepIndex = usePrevious(props.lastStepIndex);
+    // Save the last number of graphs so that we can check if a graph was deleted during an undo, 
+    // so we can update the selected tab
+    const prevNumberOfGraphs = usePrevious(Object.keys(props.graphDataJSON).length)
 
     // When the last step index changes, check if an undo occured so we can refresh the params
     useEffect(() => {
         // If there has been an undo, then we refresh the params to this pivot
         if (prevLastStepIndex && prevLastStepIndex !== props.lastStepIndex - 1) {
             void refreshParamsAfterUndo()
+
+            const graphIDs = Object.keys(props.graphDataJSON)
```

Comment:
I don't want to do this work in the Undo effect - can we handle this generally in the same place where we handle when a new graph is created?

Reply:
A lot of this logic is unique to the undoing. The complex part is figuring out if another graph exists to switch to and if not, switching to a data sheet. So I think it makes sense to not bundle together. 

But, this logic is also needed by the sheet tabs `onDelete` function, so I pulled it out into a helper function.

Reply:
Note to self: look into this logic more.